hexa/connections/dhis2_client_helper.py

```diff
@@ -0,0 +1,52 @@
+from openhexa.toolbox.dhis2 import DHIS2
+from openhexa.toolbox.dhis2.api import DHIS2Error
+
+from hexa.user_management.models import User
+from hexa.workspaces.models import Connection, ConnectionField
+
+
+def get_client_by_slug(slug: str, user: User) -> DHIS2 | None:
+    """
+    Gets DHIS2 client from workspace by its slug
+    """
+    try:
+        dhis2_connection = (
+            Connection.objects.filter_for_user(user).filter(slug=slug).first()
+        )
+    except Connection.DoesNotExist:
+        return None
+
+    if not dhis2_connection:
+        return None
+
+    fields = (
+        ConnectionField.objects.filter(connection=dhis2_connection)
+        .select_related()
+        .all()
+    )
+    dhis_connection_dict = {field.code: field.value for field in fields}
+    try:
+        return DHIS2(
+            url=dhis_connection_dict.get("url"),
+            username=dhis_connection_dict.get("username"),
+            password=dhis_connection_dict.get("password"),
+        )
+    except DHIS2Error:
+        return None
+
+
+def get_dhis2_metadata(dhis2: DHIS2, type: str, **kwargs) -> dict:
+    """
+    Gets metadata from DHIS2
+    """
+    metadata_methods = {
+        "organisation_units": dhis2.meta.organisation_units,
+        "organisation_unit_groups": dhis2.meta.organisation_unit_groups,
+        "organisation_unit_levels": dhis2.meta.organisation_unit_levels,
+        "datasets": dhis2.meta.datasets,
+        "data_elements": dhis2.meta.data_elements,
+        "data_element_groups": dhis2.meta.data_element_groups,
+        "indicator": dhis2.meta.indicators,
+        "indicator_groups": dhis2.meta.indicator_groups,
+    }
+    return metadata_methods[type](**kwargs)
```

Comment:
What's the benefit of this method ? 

It seems like an unnecessary layer (adding a metadata requires adding it here as well, string values need to be kept aligned with function names,...)

What about simply using : `dhis2.meta.data_elements(fields="id,name")` ?

Reply:
If we want to keep some dynamism ( for instance to use in `@dhis2_connection.field("query")
def resolve_query(dhis2_client, info, **kwargs):`) maybe we can explore the following approach :

```

def get_dhis2_metadata(dhis2: DHIS2, type: str, **kwargs) -> dict:
    """
    Gets metadata from DHIS2
    """
    try:
        metadata_method = getattr(dhis2.meta, type)
        return metadata_method(**kwargs)
    except AttributeError:
        raise ValueError(f"Invalid metadata type: {type}")
```

It keeps maintenance low while conserving the dynamic aspect

Reply:
I would still keep it with the method mapping. This way we could for example replace type from snake_case to camel case as it  would follow the GraphQL naming convention